backend-elasticsearch/src/main/java/org/hibernate/search/v6poc/backend/elasticsearch/document/model/impl/ElasticsearchObjectNodeModel.java

```diff
@@ -6,12 +6,15 @@
  */
 package org.hibernate.search.v6poc.backend.elasticsearch.document.model.impl;
 
+import org.hibernate.search.v6poc.backend.document.model.ObjectFieldStorage;
+
 /**
  * @author Yoann Rodiere
  */
 public class ElasticsearchObjectNodeModel {
 
-	private static final ElasticsearchObjectNodeModel ROOT = new ElasticsearchObjectNodeModel( null, null );
+	private static final ElasticsearchObjectNodeModel ROOT =
+			new ElasticsearchObjectNodeModel( null, null, null );
 
 	public static ElasticsearchObjectNodeModel root() {
```

Comment:
I didn't check locally as I have a lot of things going on in my tree but is this still useful now that you have differentiated the root index schema thing?

Reply:
Yes it is. We need to pass an ElasticsearchObjectNodeModel to contributors of the properties of the root node (non-embedded fields).